add new api docs (#56)
LemoFoundationLtd/lemo-client: README.md

---

<a name="submodule-account-createTempAddress"></a>
#### lemo.account.createTempAddress
```
lemo.account.createTempAddress(from, userId)
```
create a temp account

##### Parameters
1. `string` - account address

> 2. `string` - User id, which cannot be longer than 20 bytes in hexadecimal

Reviewer: ```suggestion
2. `string` - customized user id, which's length should be 10 bytes in hexadecimal at most
```

##### Returns
`string` - Temporary account address

##### Example
```js
const userId = '1110000000000000000'
const result = lemo.account.createTempAddress('Lemo836BQKCBZ8Z7B7N4G4N4SNGBT24ZZSJQD24D', userId)
console.log(result) // Lemo85SY56SGRTQQ63A2Y43KYA8C7QAZB37P3KY5
```

---

<a name="submodule-account-isTempAddress"></a>
#### lemo.account.isTempAddress
```
lemo.account.isTempAddress(address)
```

> True if the current address is a temporary account

Reviewer: ```suggestion
True if the specific address is a temporary account
```

const payer = 'Lemo836BQKCBZ8Z7B7N4G4N4SNGBT24ZZSJQD24D'
const noGasInfo = lemo.tx.signNoGas('0x432a86ab8765d82415a803e29864dcfc1ed93dac949abf6f95a583179f27e4bb', txInfo, payer)
const result = lemo.tx.signReimbursement('0x432a86ab8765d82415a803e29864dcfc1ed93dac949abf6f95a583179f27e4bb', noGasInfo, 2, 100)
console.log(result)
// {"type":"0","version":"1","chainID":"1","gasPrice":"2","gasLimit":"100","amount":"0","expirationTime":"1560245965","from":"Lemo836BQKCBZ8Z7B7N4G4N4SNGBT24ZZSJQD24D","to":"Lemo83JW7TBPA7P2P6AR9ZC2WCQJYRNHZ4NJD4CY","sigs":["0xf8ec13d8fb425939d00a2c97299ba57a29ae1e1fb9450e6f7a7620189d8000de1c53a9bc81e6c34138fd80dbd0959ade8e33564bdb4c9b93e9ae7edc5de7440701"],"gasPayerSigs":["0x0edb211e684bfda13969aa9115c292a485bdb43061b54148b140b97cd0322b3d7d973004d8caf0c4de3975d0d553baa6d0d7d41a6ce0d14c0b754cecb4b020a900"]}
```

---

<a name="submodule-tx-signCreateTempAddress"></a>
#### lemo.tx.signCreateTempAddress
```
lemo.tx.signCreateTempAddress(privateKey, txConfig, userId)
```
Sign a transaction for create temp account transaction, then return the signed transaction string  

> 1. Temp account have no privatekey and only can be signed by accounts in Signers

Reviewer: ```suggestion
1. Temp account has no private key and only can be signed by accounts in its `signers`
```

> 2. Temp account must be configured with Signers before they can be used

Reviewer: ```suggestion
2. Temp account must be configured with Signers before using it
```

> 3. If the temp account already exists, only multi-signed accounts can modify the Signers

Reviewer: ```suggestion
3. If the temp account already exists, the creation will be fail
```

The API is used like [`lemo.tx.sign`](#submodule-tx-sign). The only difference is filling special data in transaction

##### Parameters
1. `string` - Account private key
2. `object` - Unsigned transaction like the same parameter in [`lemo.tx.sendTx`](#submodule-tx-sendTx).

> 3. `string` - User id, which cannot be longer than 20 bytes in hexadecimal

Reviewer: ```suggestion
2. `string` - customized user id, which's length should be 10 bytes in hexadecimal at most
```

##### Returns
`string` - The string of signed [transaction](#data-structure-transaction) information

##### Example
```js
const userId = '0123456789'
const txInfo = {from: 'Lemo836BQKCBZ8Z7B7N4G4N4SNGBT24ZZSJQD24D', to: 'Lemo83JW7TBPA7P2P6AR9ZC2WCQJYRNHZ4NJD4CY'}
const result = lemo.tx.signCreateTempAddress(testPrivate, txInfo, userId)
console.log(result)
// {"type":"0","version":"1","chainID":"1","gasPrice":"3000000000","gasLimit":"2000000","amount":"0","expirationTime":"1560243152","from":"Lemo836BQKCBZ8Z7B7N4G4N4SNGBT24ZZSJQD24D","to":"Lemo85SY56SGRTQQ63A2Y48GBNCRGJC25A6HTDGR","data":"0x7b227369676e657273223a5b7b2261646472657373223a224c656d6f38333642514b43425a385a3742374e3447344e34534e47425432345a5a534a5144323444222c22776569676874223a3130307d5d7d","sigs":["0x4a807e3c5f6af4a1bd1e4ba05c7e1261bf62b8768302ab140b3c43931096f17b7fae90376f6eaed3f97c38ae2f5e83fa61c03f2683df89129469c3d8cd0df82700"],"gasPayerSigs":[]}
```

---

<a name="submodule-tx-signBoxTx"></a>
#### lemo.tx.signBoxTx
```
lemo.tx.signBoxTx(privateKey, txConfig, boxTxInfo) 
```
Sign a transaction for box transactions, then return the signed transaction string. 

> 1. Box transactions can store multiple complete transaction information including special transactions, but cannot store box transactions in box transactions

Reviewer: ```suggestion
1. Box transaction can store multiple signed transaction informations including special transactions, but cannot store box transactions
```

2. The timestamp of the box transaction is equal to the minimum timestamp of the child transaction in the box
The API is used like [`lemo.tx.sign`](#submodule-tx-sign). The only difference is filling special data in transaction

> 

Reviewer: 3. All the sub transactions in box will be success or fail at same time

##### Parameters
1. `string` - Account private key
2. `object` - Unsigned transaction like the same parameter in [`lemo.tx.sendTx`](#submodule-tx-sendTx). For this API, `to` fields will be ignored.

> 3. `object` - Box transactions stores complete transaction information

Reviewer: ```suggestion
3. `array` - Signed transaction list. The item should by signed transaction string or object
```
